idelib/dataset.py

```diff
@@ -3274,6 +3317,66 @@ def arrayResampledRange(self, startTime, stopTime, maxPoints, padding=0,
                                           display=display)
         return self.arraySlice(startIdx, stopIdx, step, display=display)
 
+    def _initializeCache(self, *args, **kwargs):
+        """ Creates a cache of the raw data, either in memory or on disk """
+        from .parsers import ChannelDataArrayBlock
```

Comment:
It is probably best to avoid imports within functions/methods unless there's a special reason for it.

Reply:
good catch